components/LanguageToggle.js

```diff
@@ -0,0 +1,149 @@
+    this.secondDaylang = document.querySelector(".second-day");
+    this.lastDaylang = document.querySelector(".last-day");
+    this.dateElement = document.querySelector(".date");
+
+    this.translations = {
+      en: {
+        overcast: "OVERCAST",
+        feelsLike: "FEELS LIKE",
+        wind: "WIND",
+        humidity: "HUMIDITY",
+        latitude: "latitude",
+        longitude: "longitude",
+        days: {
+          0: "Sunday",
+          1: "Monday",
+          2: "Tuesday",
+          3: "Wednesday",
+          4: "Thursday",
+          5: "Friday",
+          6: "Saturday",
+        },
+        months: {
+          0: "January",
+          1: "February",
+          2: "March",
+          3: "April",
+          4: "May",
+          5: "June",
+          6: "July",
+          7: "August",
+          8: "September",
+          9: "October",
+          10: "November",
+          11: "December",
+        },
+      },
+      ru: {
+        overcast: "ПАСМУРНО",
+        feelsLike: "ОЩУЩАЕТСЯ КАК",
+        wind: "ВЕТЕР",
+        humidity: "ВЛАЖНОСТЬ",
+        latitude: "широта",
+        longitude: "долгота",
+        days: {
+          0: "Воскресенье",
+          1: "Понедельник",
+          2: "Вторник",
+          3: "Среда",
+          4: "Четверг",
+          5: "Пятница",
+          6: "Суббота",
+        },
+        months: {
+          0: "Январь",
+          1: "Февраль",
+          2: "Март",
+          3: "Апрель",
+          4: "Май",
+          5: "Июнь",
+          6: "Июль",
+          7: "Август",
+          8: "Сентябрь",
+          9: "Октябрь",
+          10: "Ноябрь",
+          11: "Декабрь",
+        },
+      },
+    };
+
+    this.setupEventListeners();
+  }
+
+  setupEventListeners() {
+    this.langButton.addEventListener("click", () => {
+      this.switchLanguage();
+    });
+  }
+
+  switchLanguage() {
+    if (this.currentLanguage === "en") {
+      this.currentLanguage = "ru";
+      this.langButton.textContent = "RU";
+    } else {
+      this.currentLanguage = "en";
+      this.langButton.textContent = "EN";
+    }
+
+    this.updateLanguageText();
+  }
+
+  updateLanguageText() {
+    const translations = this.translations[this.currentLanguage];
+
+    this.weatherCondition.textContent = `${translations.overcast}`;
+    this.weatherFeelsLike.textContent = `${translations.feelsLike}: ${this.app.originalTemperatures?.feelsLike}°`;
+
+    this.weatherWind.textContent = `${translations.wind}: ${
+      this.app.originalTemperatures?.wind || "N/A"
+    } m/s`;
+
+    this.weatherHumidity.textContent = `${translations.humidity}: ${
+      this.app.originalTemperatures?.humidity || "N/A"
+    }%`;
+
+    const currentDate = new Date();
+    const dayOfWeek = translations.days[currentDate.getDay()];
+    const dayOfMonth = currentDate.getDate();
+    const month = translations.months[currentDate.getMonth()];
+    this.dateElement.textContent = `${dayOfWeek} ${dayOfMonth} ${month}`;
```

Comment:
Implement separate getCurrentDateString method, the result will be a string to insert in textContent.
```suggestion
    this.dateElement.textContent = this.getCurrentDateText();
```

Reply:
https://github.com/VasiliiDubik/Vasiliidubik-weatherForecast/pull/1/commits/486a734cbdbc0064393b279e1a945e94601d704a